azurerm/resource_arm_batch_pool.go

```diff
@@ -0,0 +1,518 @@
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vm_size": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+			"fixed_scale": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"target_dedicated_nodes": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  1,
+						},
+						"target_low_priority_nodes": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  0,
+						},
+						"resize_timeout": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "PT15M",
+						},
+					},
+				},
+			},
+			"auto_scale": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"evaluation_interval": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "PT15M",
+						},
+						"formula": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"storage_image_reference": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"publisher": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"offer": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"sku": {
+							Type:             schema.TypeString,
+							Required:         true,
+							ForceNew:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+
+						"version": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"node_agent_sku_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"stop_pending_resize_operation": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"start_task": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"command_line": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"max_task_retry_count": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  1,
+						},
+
+						"wait_for_success": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"environment": {
+							Type:     schema.TypeMap,
+							Optional: true,
+						},
+
+						"user_identity": {
+							Type:     schema.TypeList,
+							Required: true,
```

Comment:
Should this we required when all the sub properties are optional?

Reply:
I was not sure for this one: user identity is mandatory, but you should specify either a user name or an auto identity. This is why I put the `user_identity` block mandatory but the two properties inside optional. What the best way to handle that? Adding some extra validation to the `user_identity` block to check that at least one type of identity has been specified?

Reply:
If one of those values is required i would add an additional check & document it in the create/update functions for now.